compiler-core/src/docs.rs

```diff
@@ -77,20 +81,52 @@ pub fn generate_html(
         .sorted()
         .collect();
 
+    let render_as_meta_description = |content: String| -> String {
+        let content = content._strip_tags();
+        let content = manipulate::replace_all(content.as_str(), "\n", " ");
+        let content = manipulate::trim(content.as_str(), " ");
+        let multi_whitespace_regex = Regex::new(r"\s+").expect("Multi whitespace regex");
+        let content = multi_whitespace_regex.replace_all(content.as_str(), " ");
+        if content.len() <= MAX_HTML_META_DESCRIPTION_LEN {
+            return content.to_string();
+        }
+        format!(
+            "{}…",
+            manipulate::trim(
+                content.get(0..MAX_HTML_META_DESCRIPTION_LEN).unwrap_or(""),
+                " "
+            )
+        )
+    };
+
     // Generate user-supplied (or README) pages
     for page in docs_pages {
         let content = std::fs::read_to_string(&page.source).unwrap_or_default();
+        let rendered_content = render_markdown(&content);
         let unnest = page_unnest(&page.path);
 
+        let page_path_without_ext = page.path.split('.').next().unwrap_or("");
+        let page_title = match page_path_without_ext {
+            // The index page, such as README, should not push it's page title
+            "index" => format!("{} · v{}", config.name, config.version),
+            // Other page title's should say so
+            _other => format!("{} · {} · v{}", page.title, config.name, config.version),
+        };
+        let page_meta_description = match page_path_without_ext {
+            "index" => render_as_meta_description(config.description.to_string().clone()),
```

Comment:
I don't think we need to do any HTML manipulation stuff for the description field, just putting it straight in is OK